build/chip/tools.gni

```diff
@@ -12,9 +12,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import("//build_overrides/chip.gni")
+import("${chip_root}/src/crypto/crypto.gni")
 import("${chip_root}/src/platform/device.gni")
 declare_args() {
   # Build CHIP tools.
   chip_build_tools = current_os != "freertos" && current_os != "android" &&
                      chip_device_platform != "fake"
+  chip_can_build_cert_tool =
+      chip_crypto == "openssl" ||
+      (chip_crypto == "" &&
+       (current_os != "android" && current_os != "freertos" &&
+        current_os != "zephyr" && current_os != "mbed" &&
+        current_os != "webos"))
```

Comment:
This does not scale. But by now chip_crypto should not be `""` because of crypto.gni.

Suggest `chip_can_build_cert_tool = (chip_crypto != "platform")`

Reply:
The reassignment of `chip_crypto` in case it is undefined, has been moved out of `crypto.gni` and into crypto's `BUILD.gn`, so it can still be empty when `tools.gni` is evaluated. The reason I moved it out is because `crypto.gni` is now included as part of the EFR32's `args.gni` (and potentially others in the future), in order to override `chip_crypto` to set it to platform in case no specific crypto instance was provided. The issue being the environment doesn't have a definition for `current_os` at the point where `crypto.gni` gets evaluated in that chain.

I'm not a GN expert by any means, so if there is a better way to resolve that, I'm open for suggestions. I agree that having the exhaustive list of 'not these platforms' in `tools.gni` is not ideal, but it is 100% the same logic as was used to tell the top-level `BUILD.gn` whether or not to try building cert-tool.